### PR TITLE
[AKS] `az aks create/update`: Support UserAssigned Managed Identity for grafana linking in managed prometheus

### PR DESCRIPTION
**Pull request by bragi92**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

```az aks create -n kaveeshcli22 -g kaveeshcli --location westeurope --enable-azure-monitor-metrics --azure-monitor-workspace-resource-id "{full_id}" --grafana-resource-id "{full_id}"```

```az aks create -n kaveeshcli22 -g kaveeshcli --disable-azure-monitor-metrics ```

**Description**
Azure Managed Grafana now has support for UserAssigned Managed Identity and we're adding support for it in the managed prometheus addon.

**Testing Guide**
- Set context to one of the air gapped clouds
- Use the command with the grafana-resource-id parameter similar to `az aks create -n kaveeshcli22 -g kaveeshcli --location {cluster_location} --enable-azure-monitor-metrics --azure-monitor-workspace-resource-id "{full_id}" --grafana-resource-id "{full_id}"` and it should result in a a successful result. Please use a grafana resource that is creating/updated using a user assigned managed identity

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [x] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

**Review by FumingZhang — COMMENTED** [review]
lgtm

Queued [live test](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=101299894&view=results) to validate the change, test passed!

- test_aks_create_with_azuremonitormetrics
- test_aks_update_with_azuremonitormetrics
- test_aks_create_with_azuremonitorappmonitoring
- test_aks_update_with_azuremonitorappmonitoring

File: src/aks-preview/azext_aks_preview/azuremonitormetrics/amg/link.py [hunk 1]
```diff
@@ -24,7 +24,22 @@ def link_grafana_instance(cmd, raw_parameters, azure_monitor_workspace_resource_
...
         headers = ['User-Agent=azuremonitormetrics.link_grafana_instance']
         grafanaArmResponse = send_raw_request(cmd.cli_ctx, "GET", grafanaURI, body={}, headers=headers)
-        servicePrincipalId = grafanaArmResponse.json()["identity"]["principalId"]
+        # Check if 'identity' and 'type' exist in the response
+        identity_info = grafanaArmResponse.json().get("identity", {})
+        identity_type = identity_info.get("type", "").lower()
+
+        if identity_type == "systemassigned":
+            servicePrincipalId = identity_info.get("principalId")
+        elif identity_type == "userassigned":
```
Comment (yanzhudd) [inline src/aks-preview/azext_aks_preview/azuremonitormetrics/amg/link.py:33]: Do we need to convert the param into lowercase and then compare them?
```suggestion
        if identity_type.lower() == "systemassigned":
            servicePrincipalId = identity_info.get("principalId")
        elif identity_type.lower() == "userassigned":
```

**Review by FumingZhang — APPROVED** [review]

File: src/aks-preview/azext_aks_preview/azuremonitormetrics/amg/link.py (hunk 1 above)
Comment (bragi92) [reply src/aks-preview/azext_aks_preview/azuremonitormetrics/amg/link.py:33]: @yanzhudd I'm converting the parameter to lowercase on line 29. That should suffice for this comparison, I believe?

**Review by yanzhudd — APPROVED** [review]

**Pull request merged**